packages/react/src/SubNav/SubNav.tsx

```diff
@@ -176,56 +177,94 @@ type SubNavLinkProps = {
 } & PropsWithChildren<React.HTMLProps<HTMLAnchorElement>> &
   BaseProps<HTMLAnchorElement>
 
-const SubNavLink = ({
+const SubNavLinkWithSubmenu = ({
   children,
   href,
   'aria-current': ariaCurrent,
   'data-testid': testId,
   className,
   ...props
 }: SubNavLinkProps) => {
-  const hasSubMenu = Children.toArray(children).some(child => {
+  const [isExpanded, setIsExpanded] = useState(false)
+  const onFocusChange = useCallback((isFocused: boolean) => {
+    if (!isFocused) {
+      setIsExpanded(false)
+    }
+  }, [])
+  const submenuId = useId()
+  const ref = useIsChildFocused<HTMLDivElement>(onFocusChange)
+
+  const [label, SubMenuChildren] = children as ReactNode[]
+
+  const onKeyDown = useCallback((e: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (['Enter', ' '].includes(e.key)) {
+      setIsExpanded(prev => !prev)
+    }
+  }, [])
+
+  return (
+    // Disabling as the focus and blur events are handled by the useIsChildFocused hook
+    // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
```

Comment:
@rezrah Do I remember correctly that we can't disable `jsx-a11y` rules?

It's slightly annoying in this case as it's trying to get me to add `onFocus` and `onBlur` events to the div, but that's not the behaviour we want, and it wouldn't be accessible even if we did as the `div` isn't focusable.

The `useIsChildFocused` hook handles the focus/blur so we've still got all of the functionality that's required, just not in the way that this eslint rule wants.

If we can't disable the rule as I've done here, then I could either include `onBlur` and `onFocus` as noops, eg

```tsx
onFocus={noop}
onBlur={noop}
```

or attach the `onMouseOver` and `onMouseOut` handlers using `addEventListener` inside a `useEffect` as I don't think eslint will pick up on that.

Reply:
AFAIK based on my current understanding, the a11y linter uses static code analysis and there isn't a way to ignore it (which would defeat its purpose). I think you'd either have to go with noop approach like you shared. If you need to set a value, can you just set undefined or null?

Reply:
Cool, I've just set them to `onFocus={() => null}` for now and added a comment explaining the choice